protocols/mdns/src/behaviour.rs

```diff
@@ -181,55 +185,74 @@ impl Mdns {
         self.discovered_nodes.iter().map(|(p, _, _)| p)
     }
 
-    fn inject_mdns_packet(&mut self, packet: MdnsPacket, params: &impl PollParameters) {
+    fn reset_timer(&mut self) {
+        self.timeout.set_interval(self.query_interval);
+    }
+
+    fn fire_timer(&mut self) {
+        self.timeout
+            .set_interval_at(Instant::now(), self.query_interval);
```

Comment:
Given that this function does not add any randomness, wouldn't that somewhat synchronize intervals across nodes again?